### PR TITLE
Add documentation about how to customise containers, args, and tweak resources

### LINKED ISSUE #1643: Add user docs for changing `ext.args`
See https://github.com/nf-core/viralrecon/issues/348

### PR DESCRIPTION
**Pull request by jfy133**
As requested here: https://github.com/nf-core/viralrecon/issues/348

Closes #1643 

Note I also transferred over from viralrecon the containers update, and clarified a few things on the resource tweaks.

Once this is in, I'll replace the template docs to point to these docs rather than have the current wordy RNASEQ example.

<a href="https://gitpod.io/#https://github.com/nf-core/nf-co.re/pull/1646"></a>

### CODE REVIEW

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Review by maxulysse — APPROVED** [review]

File: markdown/usage/configuration.md [hunk 1]
```diff
@@ -156,27 +162,132 @@ process {
...
 > :warning: Be careful with your syntax - if you set the memory to be `200` then it will get 200 _bytes_ of memory.
 > Memory should be in quotes with a space or without quotes using a dot: `"200 GB"` or `200.GB`.
 > See the Nextflow docs for [memory](https://www.nextflow.io/docs/latest/process.html#memory),
 > [cpus](https://www.nextflow.io/docs/latest/process.html#cpus) (int) and
 > [time](https://www.nextflow.io/docs/latest/process.html#time).
 
-Usually you would save configuration such as this to a config file [as described above](#custom-configuration-files).
+If you think that the defaults in the pipeline are way off, please the pipeline developers know either on Slack or via a GitHub issue on the pipeline repository! Then we can adjust the defaults to the benefit of all pipeline users.
+
+## Updating Tool Versions
```
Comment (mashehu) [inline markdown/usage/configuration.md:199]:
```suggestion
## Updating tool versions
```

File: markdown/usage/configuration.md [hunk 2]
```diff
@@ -156,27 +162,132 @@ process {
...
+
+The [Nextflow DSL2](https://www.nextflow.io/docs/latest/dsl2.html) implementation of nf-core pipelines uses one container or conda environment per process which makes it much easier to maintain and update software dependencies.
+
+If for some reason you need to use a different version of a particular tool with the pipeline then you just need to identify the `process` name and override the Nextflow `container` or `conda` definition for that process using the `withName` declaration.
+
+For example, the [nf-core/viralrecon](https://nf-co.re/viralrecon) pipeline uses a tool called Pangolin that updates an internal database of COVID-19 lineages quite frequently. It doesn't make sense to re-release the nf-core/viralrecon everytime a new version of Pangolin has been released.
+
+In this case, a user can override the default container used by the pipeline by creating a custom config file and passing it as a command-line argument via -c custom.config.
```
Comment (mashehu) [inline markdown/usage/configuration.md:207]:
```suggestion
In this case, a user can override the default container used by the pipeline by creating a custom config file and passing it as a command-line argument via `-c custom.config`.
```

File: markdown/usage/configuration.md [hunk 3]
```diff
@@ -156,27 +162,132 @@ process {
...
+
+> **:warning: Sometimes tool developers change how tool versions are reported between updates. Updating containers may break version reporting within the pipeline and result in missing values in MultiQC version tables**
+
+## Understanding and Modifying Tool Arguments
+
+In some cases you may wish to understand which tool arguments or options a pipeline uses, or even update or change these for your own analyses.
+
+You can sometimes find out what parameters are used in a tool in by checking the longer 'help' description of different pipeline parameters, e.g. by pressing the 'help' button next to [this parameter](https://nf-co.re/funcscan/1.0.1/parameters#annotation_bakta_mincontig) in [nf-core/funcscan](https://nf-co.re/funcscan).
+
+### Finding Already Used Arguments
```
Comment (mashehu) [inline markdown/usage/configuration.md:254]:
```suggestion
### Finding already used arguments
```

File: markdown/usage/configuration.md [hunk 4]
```diff
@@ -156,27 +162,132 @@ process {
...
+
+However if this is not listed, there are two main places that a tool can have a tool argument specified.
+
+Most arguments (both mandatory or optional) are defined in the `conf/modules.conf` file in the pipeline code under the `ext.args` entry. For example, you can see the default arguments used by the `SORTMERNA` step of the nf-core/rnaseq pipeline [here](https://github.com/nf-core/rnaseq/blob/6e1e448f535ccf34d11cc691bb241cfd6e60a647/conf/modules.config#LL299).
+
+> ℹ️ Arguments specified in `ext.args` are then inserted into the module itself via the `$args` variable in the module's bash code
+
+In some cases _some_ modules have mandatory information for a tool for it to be executed, and these normally equate to 'mandatory' arguments. You can see the argument is used in the pipeline itself looking in the `script` section given module code itself, as in the pipeline's GitHub repository under `modules/<nf-core/local>/<tool>/main.nf`.
+
+### Customing Tool Arguments
```
Comment (mashehu) [inline markdown/usage/configuration.md:264]:
```suggestion
### Customising tool arguments
```

File: markdown/usage/configuration.md [hunk 5]
```diff
@@ -156,27 +162,132 @@ process {
...
+    withName: 'NFCORE_TAXPROFILER:TAXPROFILER:HOST_REMOVAL:BOWTIE2_BUILD' {
+        ext.args = "-n 0.1"
+    }
+}
+```
+
+> **:warning: It is recommended to copy and paste existing parameters in a pipelines `conf/modules.config` file, to ensure the pipeline can function as expected**
 
-If you think that the defaults in the pipeline are way off, please let us know! Then we can adjust the defaults to the benefit of all pipeline users.
+> **:warning: It is important to note updating tool parameters or changing `ext.args` comes with no warranty by the pipeline developers!**
```
Comment (mashehu) [inline markdown/usage/configuration.md:290]:
```suggestion
> **:warning: It is important to note that updating tool parameters or changing `ext.args` comes with no warranty by the pipeline developers!**
```

**Review by mashehu — APPROVED** [review]

**Pull request merged**